substrate/frame/salary/src/lib.rs

```diff
@@ -382,6 +394,54 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::tasks_experimental]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		#[pallet::task_list(vec![].into_iter())]
+		#[pallet::task_condition(|| {
+			let now = frame_system::Pallet::<T>::block_number();
+			let cycle_period = Pallet::<T, I>::cycle_period();
+			let mut status = Status::<T, I>::get().unwrap();
```

Comment:
I don't think we want to unwrap here. we should return false:
```suggestion
			let Some(mut status) = Status::<T, I>::get() else { return false };
```